src/core.ts

```diff
@@ -4,37 +4,27 @@ import Joi from "joi"
 
 import { Issue } from "./types"
 
-export const syncIssue = async ({
-  issue,
-  graphql: gql,
-  project,
-  organization,
-}: {
-  issue: Issue
-  graphql: typeof OctokitGraphQL
-  project: { number: number; targetField: string; targetValue: string }
-  organization: string
-}) => {
-  /*
-
-    Step: Fetch the project's data so that we'll be able to create an item on it
-    for the issue given as input
-
-  */
-  const projectData = await gql<{
-    organization: {
-      projectNext: {
-        id: string
-        fields: {
-          nodes: {
-            id: string
-            name: string
-            settings?: string | null
-          }[]
-        }
+type ProjectData = {
+  organization: {
+    projectNext: {
+      id: string
+      fields: {
+        nodes: {
+          id: string
+          name: string
+          settings?: string | null
+        }[]
       }
     }
-  }>(
+  }
+}
+
+const fetchProjectData: (params: {
+  gql: typeof OctokitGraphQL
+  organization: string
+  number: number
+}) => Promise<ProjectData> = async ({ gql, organization, number }) => {
```

Comment:
I don't mind if you want to do that, just bear in mind that there's a linting rule for that https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md but it's **disabled by default** on other projects, so it's not something that we should be inclined to *require* in a review - otherwise the rule would be turned on. If you think it should be turned on then feel free to start a new discussion.

Reply:
I agree with you. This is something that I feel like doing, and I personally feel like it saves me time, but I'd like that to stay an option, rather than becoming a requirement.